### PR TITLE
Tester

### PR DESCRIPTION
**Pull request by g-ortuno**
First proof of concept for Certification App. Please review @schilit 
tester_demo

### CODE REVIEW

File: android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/Tests.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.ListIterator;
+import java.util.UUID;
+
+public class Tests {
+
+  private static final String TAG = Tests.class.getCanonicalName();
+
```
Comment (schilit) [inline android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/Tests.java]: use other version

File: android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/TestsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          Log.d(TAG, "Type: " + action.actionType + " Number: " + i);
+          holder.mTestResult.setText("#" + (i + 1) + ". " + action.reason);
+          holder.mTestResult.setVisibility(View.VISIBLE);
+          break;
+        }
+      }
+    }
+  }
+
+  private void setIcon(ViewHolder holder, UriBeaconTestHelper test) {
```
Comment (schilit) [inline android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/TestsAdapter.java]: image with states

File: android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/UriBeaconTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  private BluetoothGattService mService;
+  private String mName;
+  private Context mContext;
+  private ScanResult mResult;
+  private UUID mServiceUuid;
+  private BluetoothGattCallback mGattCallback = new BluetoothGattCallback() {
+    @Override
+    public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState) {
+      super.onConnectionStateChange(gatt, status, newState);
+      Log.d(TAG, "Status: " + status);
```
Comment (schilit) [inline android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/UriBeaconTestHelper.java]: combine into one line

File: android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/TestsAdapter.java (hunk 2 above)
Comment (g-ortuno) [inline android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/TestsAdapter.java]: Adding custom states would require quite a bit of [code](http://stackoverflow.com/questions/4336060/android-how-to-add-a-custom-button-state). I don't think the final result is worth it.

File: android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/UriBeaconTestHelper.java (hunk 3 above)
Comment (g-ortuno) [inline android-uribeacon/uribeacon-validator/src/main/java/org/uribeacon/validator/UriBeaconTestHelper.java]: Done.

**Comment by g-ortuno** [comment]
Ready for review.

**Pull request merged**